src/main/java/teammates/logic/api/TaskQueuer.java

```diff
@@ -294,6 +294,21 @@ public void scheduleEmailsForSending(List<EmailWrapper> emails) {
         }
     }
     
+    /**
+     * Schedules production of searchable documents
```

Comment:
That's a good point. If we go that way we'll probably need 'try repeatedly every x seconds until the expected data appears or a max wait time has passed, whichever is earlier' function.  
On the other hand this PR creates an entirely new queue that improves the performance only in a minor way (2s per request?) and also makes one part of the operation untestable (i.e. the search part). So I'm not sure if it should be merged (it can count as a contribution nevertheless). What do you guys think? Put this on hold and try the feasibility of a better solution first? e.g.

1. Put the entire op in the task queue in a testable way
2. Fix bottlenecks in lower layers such that the entire app can benefit from those improvements